test/test_tensor_creation_ops.py

```diff
@@ -4033,6 +4033,27 @@ def check(**kwargs):
         check(requires_grad=False)
         check(requires_grad=False, copy=True)
 
+    @onlyCPU
+    def test_astensor_consistency(self, device):
```

Comment:
Cool test

Reply:
Let's add a comment referencing the issue this PR fixes so people know what this test is doing and can see the history of why it was added